rasr/command.py

```diff
@@ -118,19 +123,53 @@ def select_exe(cls, specific_exe, default_exe_name):
             return cls.default_exe(default_exe_name)
         return specific_exe
 
-    def run_script(self, task_id, log_file, cmd="./run.sh", args=None, retries=2):
+    def run_script(
+        self,
+        task_id: int,
+        log_file: Union[str, Path],
+        cmd: str = "./run.sh",
+        args: List = None,
+        retries: int = 2,
+        use_tmp_dir: bool = False,
+    ):
         args = [] if args is None else args
         tmp_log_file = remove_suffix(
             os.path.basename(tk.uncached_path(log_file)), ".gz"
         )
-        self.run_cmd(cmd, [task_id, tmp_log_file] + args, retries)
+
+        if use_tmp_dir or (
+            hasattr(gs, "JOB_RUN_RASR_TMP_DIR") and gs.JOB_RUN_RASR_TMP_DIR
+        ):
+            with tempfile.TemporaryDirectory(prefix=gs.TMP_PREFIX) as tmp_dir:
+                print("using temp-dir: %s" % tmp_dir)
+                try:
+                    self.run_cmd(cmd, [task_id, tmp_log_file] + args, retries, tmp_dir)
+                    file_names = os.listdir(tmp_dir)
+                    for fn in file_names:
+                        shutil.move("%s/%s" % (tmp_dir, fn), fn)
```

Comment:
Does this string always yield proper directory names regarding / etc.? This is more of a beauty aspect, that the name is rather e.g. logs/log_1 and not logslog_1

Reply:
I think this should be ok. but I did not test this. @Atticus1806 you ran this, does this cause problems for you?

Reply:
we could to `os.path.join` instead of `"%s/%s" %`, which would then be more general over other os.. but internally this is anyway what's happening.

Or are you afraid of double `"//"`? I don't think it would be a problem.